src/libmongoc/src/mongoc/mongoc-client-session-private.h

```diff
@@ -83,6 +84,9 @@ struct _mongoc_client_session_t {
    uint32_t client_generation;
    uint32_t server_id;
    bson_t *recovery_token;
+   uint32_t snapshot_time_timestamp;
+   uint32_t snapshot_time_increment;
+   bool snapshot_time_set;
```

Comment:
This was my best guess at an optional timestamp 🤷 . I could have used pointers to `uint32_t`s, but the underlying values kept getting lost as I passed them around. Let me know what you think (I use the same structure for `at_cluster_time` in `read_concern`).